### ISSUE TITLE
gpu-mon requires gtk modules even when --gui is not used

### ISSUE THREAD

**Issue by min-xu-ai** [Ricks-Lab/gpu-utils#134]
on my system, I don't have the required gtk python module installed. I only need gpu-mon to run in text mode. However, there are imports as well as dependences on Gtk.Window class even when --gui is not specified. It would be nice if the text mode monitor can work without the dependencies on gtk (and gi) modules.

**Comment by Ricks-Lab** [comment]
Thanks for the suggestion.  Let me take a look at it after I get things ready for next Debian release.

**Comment by Ricks-Lab** [comment]
I did a proof of concept for this that worked.  I need to work out a more robust solution after I complete work on release of v3.8.0.  Might be a few weeks before I get to it.

**Comment by min-xu-ai** [comment]
Thank you so much! I can help test it when you have it. I have a system with bare minimal linux install, which doesn't have the needed libraries for gui.

**Comment by Ricks-Lab** [comment]
I have a working version on the `gtk_optional` branch.  Let me know of any issues.

**Comment by min-xu-ai** [comment]
Cool. I tried on my system. Here is the diff I needed to make it work:

```
diff --git a/gpu-mon b/gpu-mon
index 8b291d0..0366535 100755
--- a/gpu-mon
+++ b/gpu-mon
@@ -65,8 +65,9 @@ try:
     import gi
     gi.require_version('Gtk', '3.0')
     from gi.repository import GLib, Gtk
+    from GPUmodules import GPUgui
     GTK = True
-except ModuleNotFoundError as error:
+except (ModuleNotFoundError, ValueError) as error:
     print('gi import error: {}'.format(error))
     print('gi is required for {}'.format(__program_name__))
     print('   In a venv, first install vext:  pip install --no-cache-dir vext')
@@ -74,7 +75,6 @@ except ModuleNotFoundError as error:
     GTK = False

 from GPUmodules import __version__, __status__, __credits__
-from GPUmodules import GPUgui
 from GPUmodules import GPUmodule as Gpu
 from GPUmodules import env
```

The ValueError is from gi.require_version() line since gi could be install but underlying gtk module can be not there. without it, the error is:

```
Traceback (most recent call last):
  File "/home/owen/e/py39/bin/gpu-mon", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/home/owen/git/gpu-utils/gpu-mon", line 66, in <module>
    gi.require_version('Gtk', '3.0')
  File "/home/owen/miniconda39/lib/python3.9/site-packages/gi/__init__.py", line 126, in require_version
    raise ValueError('Namespace %s not available' % namespace)
ValueError: Namespace Gtk not available
```

**Comment by Ricks-Lab** [comment]
Can you make a pull request with the modifications?

**Comment by min-xu-ai** [comment]
Sure. See #135. I just moved the GPUgui import into the same try statement. That should be the same, right?

**Comment by Ricks-Lab** [comment]
Looks good!  I made another change to simplify the `if GTK` logic.  Also, I updated credits to indicate your contribution.  Let me know of any concerns.

**Comment by Ricks-Lab** [comment]
Enhancement is complete and will be released as a package in v3.8.1.  Need to delay this to support v3.8.0 official Debian package release.
